tutel/gates/cosine_top.py

```diff
@@ -28,7 +28,6 @@ def forward(self, x):
             sim_matrix = self.sim_matrix
         logits = torch.matmul(F.normalize(cosine_projector(x), dim=1),
                               F.normalize(sim_matrix, dim=0))
-        logit_scale = torch.clamp(self.temperature, max=torch.log(torch.tensor(1. / 0.01))).exp()
         logit_scale = torch.clamp(self.temperature, max=torch.log(torch.tensor(1. / 0.01)).to(x.device)).exp()
```

Comment:
Thanks! What about putting device casting after `exp()`?

Reply:
Even that might throw errors, since `self.temperature`  would be on device `cuda` (if `CosineTopKGate` is casted to cuda) and the second argument of `torch.clamp`, `torch.log(torch.tensor(1. / 0.01))` would be on cpu.

Reply:
OK, can you replace it with 4.605170185988092? Since it is a constant value not needed to be calculated every time. Thus, casting to device is not necessary.

Reply:
Alright! I have initialized a new member variable `self.clamp_max` and used that instead. Used `item()` to get the constant(rather than tensor) during initialization.